companion/gateway/database.py

```
@@ -0,0 +1,535 @@
+        
+        try:
+            self.pool = await asyncpg.create_pool(
+                self.database_url,
+                min_size=self.settings.db_pool_min_size,
+                max_size=self.settings.db_pool_max_size,
+                command_timeout=60
+            )
+            self._initialized = True
+            logger.info("Database connection pool initialized successfully")
+        except Exception as e:
+            logger.error(f"Failed to initialize database connection pool: {e}")
```

Ruff (0.13.2)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)